File: app/src/main/java/closer/vlllage/com/closer/handler/group/PhotoActivityTransitionHandler.java
package closer.vlllage.com.closer.handler.group;

import android.content.Intent;
import android.graphics.Rect;
import android.view.View;

import closer.vlllage.com.closer.PhotoActivity;
import closer.vlllage.com.closer.handler.helpers.ActivityHandler;
import closer.vlllage.com.closer.handler.helpers.ApplicationHandler;
import closer.vlllage.com.closer.pool.PoolMember;

import static closer.vlllage.com.closer.PhotoActivity.EXTRA_PHOTO;

public class PhotoActivityTransitionHandler extends PoolMember {
    public void show(View view, String photo) {
        Intent intent = new Intent($(ApplicationHandler.class).getApp(), PhotoActivity.class);
        intent.setAction(Intent.ACTION_VIEW);
        intent.putExtra(EXTRA_PHOTO, photo);

        if (view != null) {
            Rect bounds = new Rect();
            view.getGlobalVisibleRect(bounds);

            intent.setSourceBounds(bounds);
        }

        $(ActivityHandler.class).getActivity().startActivity(intent);
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/group/PhotoActivityTransitionHandler.kt
package closer.vlllage.com.closer.handler.group

import android.content.Intent
import android.graphics.Rect
import android.view.View
import closer.vlllage.com.closer.PhotoActivity
import closer.vlllage.com.closer.PhotoActivity.Companion.EXTRA_PHOTO
import closer.vlllage.com.closer.handler.helpers.ActivityHandler
import closer.vlllage.com.closer.handler.helpers.ApplicationHandler
import com.queatz.on.On

class PhotoActivityTransitionHandler constructor(private val on: On) {
    fun show(view: View?, photo: String) {
        val intent = Intent(on<ApplicationHandler>().app, PhotoActivity::class.java)
        intent.action = Intent.ACTION_VIEW
        intent.putExtra(EXTRA_PHOTO, photo)

        if (view != null) {
            val bounds = Rect()
            view.getGlobalVisibleRect(bounds)

            intent.sourceBounds = bounds
        }

        on<ActivityHandler>().activity!!.startActivity(intent)
    }
}
